docs/developers/wallet-integration.md

```diff
@@ -0,0 +1,279 @@
+
+When a new vault is created with a factory, the factory emits a `NewVault` event to the logs that includes the vault address and the address of the underlying asset in the vault (i.e. USDC).
+
+```solidity
+NewVault (index_topic_1 address vault_address, index_topic_2 address asset)
+```
+
+Each new version of Vaults that are deployed will have its own factory, and each factory will have its own logs of events, so to get all of the deployed vaults you will need to iterate over all the factories.
+
+### Querying Registries
+
+Since V3 factories are permissionless, Yearn also deploys permissioned registry contracts that only Yearn governance can add to. These are the best place to find the official (endorsed) Yearn Vaults that are deployed by Yearn. As mentioned above, the most recent registry will resolve from registry.v3.ychad.eth. But you will need to query all the registries to get all the endorsed vaults.
+
+To get the endorsed vaults, you can either:
+
+**A.** Search through the Events Log of the registry contract and save all `NewEndorsedVault` events. This data contains the vault address, the underlying asset address, the release Version, and the Vault Type.
+
+```solidity
+NewEndorsedVault (index_topic_1 address vault, index_topic_2 address asset, uint256 releaseVersion, uint256 vaultType)
+```
+
+**B.** Or query the registry factory directly. To do that, you need to:
+
+1. Call the `getAssets()` function to get an array of underlying tokens with deployed vaults. This array will be used in the next step.
+2. Loop through each entry in the array and call the `getEndorsedVaults()` function with the address value from the array as the argument to get an array of vaults for each underlying asset
+
+```js title="Example pseudo-code"
+const allVaultsFromRegistry = []
+const assetArray = getAssets()
+for (asset in assetArray) {
+    const vaultsArray = getEndorsedVaults(asset)
+    allVaultsFromRegistry.push(...vaultsArray)
+}
+```
+
+Additional vault data can then be found by calling the respective read functions on the vault contracts (`name()`, `symbol()`, `decimals()`, `token()`, `pricePerShare()`, etc.).
+
+Like the factories, you will need to query all registry contracts to get all the endorsed vaults.
+
+## Getting V2 yVaults Manually
+
+Below are the relevant registries for V2 contracts.
+
+| Registry / ENS | Chain (ID) | Contract Address |
+|----------------|------------|------------------|
+| Mainnet Factory Registry | 1 | 0xaF1f5e1c19cB68B30aAD73846eFfDf78a5863319 |
+| Mainnet Legacy Registry 1 | 1 | 0xe15461b18ee31b7379019dc523231c57d1cbc18c |
+| Mainnet Legacy Registry 2 | 1 | 0x50c1a2eA0a861A967D9d0FFE2AE4012c2E053804 |
+| Optimism Legacy Registry | 10 | 0x1ba4eB0F44AB82541E56669e18972b0d6037dfE0 |
```

Comment:
There are factory/release registries on OP, Arbitrum, and Base too

OP
factory: 0x79286Dd38C9017E5423073bAc11F53357Fc5C128
release: 0x81291ceb9bB265185A9D07b91B5b50Df94f005BF

arbitrum
factory: 0x84654e35E504452769757AAe5a8C7C6599cBf954
release: 0x697BC6bd64677bE63240262869dD5F2A3eEACCd3

base
factory: 0xF3885eDe00171997BFadAa98E01E167B53a78Ec5
release: 0x697BC6bd64677bE63240262869dD5F2A3eEACCd3

Reply:
also fwiw, other ENS

0xaF1f5e1c19cB68B30aAD73846eFfDf78a5863319: v2.registry.ychad.eth, factory.ychad.eth
0x7Cb5ABEb0de8f6f46a27329B9eF54CE10E47F1e2: release.registry.ychad.eth

Reply:
Cool. Will update. But wasn't planning on adding the release registries here since they aren't needed to find the vaults. Will add a link to the addresses page for other contracts.